src/fixate/drivers/dmm/fluke_8846a.py

```diff
@@ -101,6 +111,25 @@ def measurements(self):
         with self.lock:
             return self._read_measurements()
 
+    def min_avg_max(self, samples=1, sample_time=1):
```

Comment:
I personally don't think we should be adding math functions into the DMM driver. We already have the ability to take a whole bunch of measurements, we can do whatever statistical analysis we want in the test script. 
I am happy to be overruled on this by @clint-lawrence or @jcollins1983 but that's just my opinion. 

The way we currently have to sample a bunch of measurements uses the manual triggering functions:
```
dmm.set_manual_trigger(samples=100) # Get ready to take 100 samples
dmm.trigger() # Start taking samples
values = dmm.measurements() # Get a list of the measurements
```
I have realised that the ``trigger()`` function (at least for the Keithley) will give a timeout error if you set a very large number of samples and don't increase ``instrument.timeout``. Maybe we just need to improve that function to wait for the samples to be taken, by querying some status register on the DMM.

Reply:
> I personally don't think we should be adding math functions into the DMM driver

I'm generally in agreement. Unless it someone we do lots of and perhaps its worth the effort to standardise. But in that case, have a generic function shared by the different driver would be preferable to having to implement on different drivers.

Other option open to us are:
- Make sure any common patterns are added to documentation, so they don't need to be re-discovered
- Create a module of utility functions that capture common pattern, but aren't tied to a specific driver

> I have realised that the trigger() function (at least for the Keithley) will give a timeout error if you set a very large number of samples and don't increase instrument.timeout

This has been a pain in the past. As best I can tell, the semantics around how to actually use SCPI command efficiently is inconsistent across instruments and poorly documented. I'm sure there is at least one place where we adjust the timeout on the fly to achieve this.

Reply:
I tried using the trigger() function in the keitley driver for this and did run into that timeout issue.

The way I think about it is that the DMMs have the ability to do these calculations in the device so it makes sense to be able to use them. It's not that we're doing the calculations in python in the driver.

Reply:
>> I personally don't think we should be adding math functions into the DMM driver

>I'm generally in agreement. Unless it someone we do lots of and perhaps its worth the effort to standardise. But in that case, have a generic function shared by the different driver would be preferable to having to implement on different drivers.

I agree with both statements here.

Reply:
> > > I personally don't think we should be adding math functions into the DMM driver
> 
> > I'm generally in agreement. Unless it someone we do lots of and perhaps its worth the effort to standardise. But in that case, have a generic function shared by the different driver would be preferable to having to implement on different drivers.
> 
> I agree with both statements here.

The motivator behind making this change is to simplify the test script drastically as well as to allow it to work on both the keithley and fluke.
![image](https://github.com/user-attachments/assets/ecfa11b3-fcde-475a-8ff0-89085d2a3668)

Reply:
@Overlord360 Thanks, that is a great example to add to the PR description, so reviewers understand the context a little better. I was trying to avoid getting too sucked into reviewing this, but since I'm here a few thoughts...

`set_nplc` would make a good context manager
```python
with dmm.set_nplc(nplc):
    return dmm.measure()
```
If you don't like that, then perhaps return the old value from `set_nplc`. Then it can be stored locally and passed back into the second call. That is a fairly common pattern for API that changes/reset some kind of global state. But a context manager is better because it is more reliable in the event of exception being raise

Avoid prefixing builtin names with an underscore. That convention is typically used to indicate something as private in python. Instead you can use a postfix underscore.

Or even better, return a class with appropriate attributes. Saves the awkward naming and is easier to consume

```python
# in some common file
@dataclass
class MeasurementStats:
    min: float
    max: float
    mean: float
```

Then you can return that class directly from the dmm function.

See the example and understanding that this tidies up existing usage, I'm happy to see this get included.

Reply:
I've added a context manager for setting nplc in helper.py. 

I've also made the min_avg_max function return a dictionary of the results with the keys: "min", "avg" and "max" as the class that Clint suggested is pretty much just a dictionary.

I've also added some tests for these new functions.

Reply:
> I've also made the min_avg_max function return a dictionary of the results with the keys: "min", "avg" and "max" as the class that Clint suggested is pretty much just a dictionary.

No, I disagree with this. There are a couple of important differences.

It's useful to think of a dataclass as a "record" or "struct" data type; A grouping of members. The significant thing is we know what the members are (`min`, `max` and `avg`). The makes it more discoverable and easier to use at the call site as most IDEs will be able to analyse the code and know the return type of the method and the available attributes on that type. With a dictionary, the keys are completely unconstrained.

The second difference (which is mostly a consequence of the first) is that returning dict with a of known keys is uncommon in python code. While that shouldn't rule it out from consideration, to go against the common practise I expect there to be a strong justification.

One minor thing that I did above but didn't specifically call out. I replaced `avg` with `mean` since it is more specific. I'm not too bother if it stays as `arg`, since it is fairly common for people to assume mean when the method of calculating an average isn't specified.

Reply:
This has been done as of 4bd69b0